backend/routes/goals.py

```
@@ -278,12 +271,50 @@ def should_show_personal_goal_button():
         current_date = datetime.now().date()
         if current_date.day <= 7:
             return jsonify({"showButton": True}), 200
-        
+
         return jsonify({"showButton": False}), 200
 
     except Exception as e:
-        logger.error(f"An error occurred while checking allow personalized goals, {str(e)}")
+        logger.error(
+            f"An error occurred while checking allow personalized goals, {str(e)}"
+        )
```

Ruff (0.8.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)